src/application/App.tsx

```diff
@@ -36,97 +84,125 @@ const GET_OPERATION_COUNTS: TypedDocumentNode<
 `;
 
 export const App = () => {
+  const mountedRef = useRef(false);
   const [settingsOpen, setSettingsOpen] = useState(false);
   const { data } = useQuery(GET_OPERATION_COUNTS);
+  const [clientNotFoundModalOpen, setClientNotFoundModalOpen] = useState(false);
   const selected = useReactiveVar<Screens>(currentScreen);
-  const reloading = useReactiveVar<boolean>(reloadStatus);
+  const state = useReactiveVar(devtoolsState);
   const [embeddedExplorerIFrame, setEmbeddedExplorerIFrame] =
     useState<HTMLIFrameElement | null>(null);
 
-  // During a reload, reset the current screen to Queries.
   useEffect(() => {
-    if (reloading && selected !== Screens.Queries) {
-      currentScreen(Screens.Queries);
+    let timeout: NodeJS.Timeout;
+    // Don't show connected message on the first render if we are already
+    // connected to the client.
+    if (!mountedRef.current && state === "connected") {
+      return;
     }
-  }, [reloading, selected]);
 
-  if (reloading) {
-    return <div></div>;
```

Comment:
I also discovered that it would empty out the screen anytime a page reload happens. I opted to remove this and keep the existing UI up while the page is being reloaded. The user will see a "disconnected" message in the bottom instead until the page has loaded and the client instance is detected again.